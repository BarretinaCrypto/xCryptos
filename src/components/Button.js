import { array, bool, func, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { Text, View } from 'react-native';
import Touchable from './Touchable';
import styles from './Button.style';

const Button = ({ caption, captionStyle, disabled, onPress, style }) => (
  <Touchable onPress={!disabled ? onPress : undefined}>
    <View style={[styles.container, style, (disabled ? styles.disabled : undefined)]}>
      <Text style={[styles.caption, captionStyle]}>{caption}</Text>
    </View>
  </Touchable>
);

Button.propTypes = {
  caption: string,
  captionStyle: oneOfType(array, number),
  disabled: bool,
  onPress: func,
  style: oneOfType(array, number),
};

Button.defaultProps = {
  caption: undefined,
  captionStyle: [],
  disabled: false,
  onPress: undefined,
  style: [],
};

export default Button;
