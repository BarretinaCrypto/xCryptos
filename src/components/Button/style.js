import { StyleSheet } from 'react-native';

import { THEME } from '../../config';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: THEME.ACCENT,
    borderRadius: THEME.OFFSET * 2,
    display: 'flex',
    paddingVertical: THEME.UNIT,
    paddingHorizontal: THEME.OFFSET,
  },

  disabled: {
    backgroundColor: THEME.FONT_PRIMARY_COLOR,
    opacity: 0.35,
  },

  caption: {
    color: THEME.WHITE,
    fontSize: THEME.FONT.SIZE.NORMAL,
    fontWeight: THEME.FONT.WEIGHT.BOLD,
  },
});
