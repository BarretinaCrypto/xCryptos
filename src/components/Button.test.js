import 'react-native';
import React from 'react';
import Button from './Button';
import renderer from 'react-native-test-utils';

jest.mock('TouchableOpacity', () => 'TouchableOpacity');

it('renders by default', () => {
  const tree = renderer(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a caption', () => {
  const tree = renderer(<Button caption="xCryptos rocks!" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls onPress prop when pressed', () => {
  const onPress = jest.fn();
  const component = renderer(<Button onPress={onPress} />);

  component.query('TouchableOpacity').simulate('press');

  expect(onPress).toHaveBeenCalled();
});

describe('when disabled', () => {
  it('renders with disabled stlyes', () => {
    const tree = renderer(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not attach onPress if disabled', () => {
  const onPress = jest.fn();
  const component = renderer(<Button disabled onPress={onPress} />);

  expect(component.query('TouchableOpacity').props.onPress).toBeUndefined();
});
});

