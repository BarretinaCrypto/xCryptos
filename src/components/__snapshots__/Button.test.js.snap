// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a caption 1`] = `
<TouchableOpacity
  onPress={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed1e79",
          "borderRadius": 32,
          "display": "flex",
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        },
        Array [],
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          Array [],
        ]
      }
    >
      xCryptos rocks!
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`renders by default 1`] = `
<TouchableOpacity
  onPress={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed1e79",
          "borderRadius": 32,
          "display": "flex",
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        },
        Array [],
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          Array [],
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`when disabled renders with disabled stlyes 1`] = `
<TouchableOpacity
  onPress={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed1e79",
          "borderRadius": 32,
          "display": "flex",
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        },
        Array [],
        Object {
          "backgroundColor": "black",
          "opacity": 0.35,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          Array [],
        ]
      }
    />
  </View>
</TouchableOpacity>
`;
